policy/modules/system/systemd.te

```diff
@@ -1417,3 +1417,516 @@ optional_policy(`
+files_manage_isid_type_dirs(systemd_homed_t)
+# AVC avc:  denied  { read } for  pid=1000 comm="systemd-homewor" name=".identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { open } for  pid=1000 comm="systemd-homewor" path="/run/systemd/user-home-mount/testuser/.identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { getattr } for  pid=1000 comm="systemd-homewor" path="/run/systemd/user-home-mount/testuser/.identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
+files_manage_isid_type_files(systemd_homed_t)
+
+#### Allow: manage .identity after labels are fixed
+# AVC avc:  denied  { create } for  pid=1000 comm="systemd-homewor" name=".#.identity9bbf460fb708fc96" scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:user_home_dir_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { getattr } for  pid=1000 comm="systemd-homewor" path="/home/testuser/.#.identity9bbf460fb708fc96" dev="dm-15" ino=137 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:user_home_dir_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { open } for  pid=1000 comm="systemd-homewor" path="/run/systemd/user-home-mount/testuser/.identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { read } for  pid=1000 comm="systemd-homewor" name=".identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { rename } for  pid=1000 comm="systemd-homewor" name=".#.identity9bbf460fb708fc96" dev="dm-15" ino=137 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:user_home_dir_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { setattr } for  pid=1000 comm="systemd-homewor" name=".#.identity9bbf460fb708fc96" dev="dm-15" ino=137 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:user_home_dir_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { unlink } for  pid=1000 comm="systemd-homewor" name=".identity" dev="dm-15" ino=136 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=unconfined_u:object_r:user_home_t:s0 tclass=file permissive=1
+# AVC avc:  denied  { write open } for  pid=1000 comm="systemd-homewor" path="/home/testuser/.#.identity9bbf460fb708fc96" dev="dm-15" ino=137 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:user_home_dir_t:s0 tclass=file permissive=1
+manage_files_pattern(systemd_homed_t, user_home_dir_t, user_home_dir_t)
```

Comment:
I would probably give that ~/.identity file a private home type. Unfortunately you cannot use name-based type transitions here as this file gets created with a random name and then renamed later. Although then again, homework probably uses setfscreatecon here so it would label it according to the spec manually.